packages/@o3r/store-sync/src/sync-storage/storage-sync.ts

```diff
@@ -328,7 +331,7 @@ export const syncStorage = (config: SyncStorageConfig) => (reducer: any) => {
 
     // If state arrives undefined, we need to let it through the supplied reducer
     // in order to get a complete state as defined by user
-    if (action.type === INIT && !state) {
+    if ((action.type === INIT || action.type === UPDATE || action.type === RECOMPUTE) && !state) {
```

Comment:
`UPDATE` and `RECOMPUTE` should trigger the reducer only if no `INIT` has been emitted before for this store, isn't it?

Reply:
I'm not sure if I well understood
I think he need to call the reducer to have the initialState of a lazyloaded store 
or maybe the change should be done a bit above in this file [L269](https://github.com/AmadeusITGroup/otter/blob/main/packages/%40o3r/store-sync/src/sync-storage/storage-sync.ts#L269)

Reply:
*Turned this comment as unresolved, please do not resolve a question without the question author approval*

If we agree on the fact that the `reducer` should be called on `RECOMPUTE` and `INIT` actions *only* if the `INIT` event has not be emitted for this store, this is not the implementation on the code.
So:
- either you should consider checking that the `INIT` action has not been emitted to this store before checking for the 2 others actions,
- either it s a case it never happen (then an explanation as response of the question would be appreciated), 
- either executing the `reducer` a second time on a store that has already got an `INIT` action is not an issue (an explanation would also be needed).

Reply:
As mentioned above, when syncing the state using the `syncStorage` method within a lazily loaded feature module, the `INIT` action is not received and therefore, the reducer is not invoked resulting in an undefined state.

So, point no 2 from your comment. `INIT` is never called.

Please check the minimal reproduction of the same: https://github.com/Abhirocks889/store-sync-error